### DISCUSSION TITLE
使用 event.get_message() 时不会返回命令所带的参数，而是返回全部

### DISCUSSION THREAD

**Discussion by EthanolM** [nonebot/discussions#170] [问答]
例如输入 /天气 上海    
根据教程

`args = str(event.get_message()).strip()  # 首次发送命令时跟随的参数，例：/天气 上海，则args为上海`

则此时args应为上海

但在我这里args则为

> /天气 上海

也就是把命令也给带上了
不知道改如何解决

以下为代码
```
from nonebot import on_command
from nonebot.rule import to_me
from nonebot.typing import T_State
from nonebot.adapters import Bot, Event

search = on_command("查询",priority=5)

@search.handle()
async def search_receive(bot: Bot, event: Event, state: T_State):
    args = str(event.get_message()).strip()
    print(args)
    await search.finish(args)  # 首次发送命令时跟随的参数，例：/天气 上海，则args为上海
```

**Comment by MingxuanGame** [comment]
你应该是安装了新版 nonebot2，但是看的是旧版文档

新版文档：https://v2.nonebot.dev

---

在新版本中，`on_command` 获取的 message 会带有命令（[Discussion 74](https://github.com/nonebot/discussions/discussions/74#discussion-3783847)）

你可以使用 [`CommandArg`](https://v2.nonebot.dev/docs/tutorial/plugin/create-handler#commandarg) 这个依赖注入拿到当前的命令参数
